core/flyout_button.js

```diff
@@ -162,6 +162,8 @@ Blockly.FlyoutButton.prototype.onMouseUp = function(e) {
   e.preventDefault();
   // Don't propagate mousewheel event (zooming).
   e.stopPropagation();
-
+  // Stop binding to mouseup and mousemove events--flyout mouseup would normally
+  // do this, but we're skipping that.
+  Blockly.Flyout.terminateDrag_();
```

Comment:
This is fine since it fixes the bug and whatnot and it undoes what needs to be undone. 

Higher level questions though: should mousedown on the flyout button call the flyout's onmousedown? Mousedown on the blocks in the flyout doesn't seem to have this behavior.

Also, after playing around with this a little, I wonder if we should change this to a click rather than a mouse up and mouse down. I can mousedown on the background of the flyout, move over to the button and then mouse up over the button and it'll create a new variable.  Seems unexpected.